tensornetwork/network_components.py

```diff
@@ -638,6 +642,27 @@ def get_tensor(self) -> Tensor:
   def set_tensor(self, tensor) -> None:
     self.tensor = tensor
 
+  def copy(self) -> "Node":
+    new_node = Node(self.tensor,
+                    name=self.name, 
+                    axis_names=self.axis_names, 
+                    backend=self.backend)
+    visited_edges = set()
+    for i, edge in enumerate(self.edges):
+      if edge in visited_edges:
+        continue
+      visited_edges.add(edge)
+      if edge.node1 == edge.node2:
+        new_edge = Edge(new_node, i, 
+                        name=edge.name,
+                        node2=new_node, 
+                        axis2=edge.axis2)
+        new_node.add_edge(new_edge, i)
+        new_node.add_edge(new_edge, edge.axis2)
+      else:
+        new_node.add_edge(Edge(new_node, i, name=edge.name), i)
+    return new_node
+
```

Comment:
Might make sense to add a `conjugate` so that you can do `node.copy(conjugate=True)`.

Reply:
Do we want this to apply to the CopyNode also? Or should CopyNode ignore a conjugate argument?

Reply:
Add the argument so the function is compatible, but a copy node is totally real, so a conjugation is a no-op. Just ignore it.